engine/src/main/scala/cromwell/webservice/CromwellApiService.scala

```diff
@@ -250,6 +250,8 @@ trait CromwellApiService extends HttpInstrumentation {
           case Success(WorkflowStoreEngineActor.WorkflowOnHoldToSubmittedFailure(_, e: NotInOnHoldStateException)) => e.errorRequest(StatusCodes.Forbidden)
           case Success(WorkflowStoreEngineActor.WorkflowOnHoldToSubmittedFailure(_, e)) => e.errorRequest(StatusCodes.InternalServerError)
           case Success(r: WorkflowStoreEngineActor.WorkflowOnHoldToSubmittedSuccess) => completeResponse(StatusCodes.OK, toResponse(r.workflowId, WorkflowSubmitted), Seq.empty)
+          case Failure(e: UnrecognizedWorkflowException) => e.failRequest(StatusCodes.NotFound)
+          case Failure(e: InvalidWorkflowException) => e.failRequest(StatusCodes.BadRequest)
```

Comment:
ToL: Since these are presumably also relevant in other response handlers, could we wrap these "standard error catchers" into a `PartialFunction` and use the `orElse` operator to include them in all of the relevant handlers?